addons/sale_expense/tests/test_reinvoice.py

```diff
@@ -8,243 +8,376 @@
 
 @tagged('-at_install', 'post_install')
 class TestReInvoice(TestExpenseCommon, TestSaleCommon):
+    """
+    Test that expenses, when linked to a sale order and invoiced are correctly re-invoiced on the sale order.
+    It should cover the following rules:
+        - Lines are never grouped together (even if re-invoiced at sale price and with a re-invoice delivered policy)
+        - When posting the move of an expense, it creates the corresponding SOLs with the correct expense quantity
+        - The amount of analytic account linked do not impact the quantities on the SOLs
+        - The quantities ordered and delivered are reset to 0 when:
+            - the expense sheet move has been reset to draft
+            - the expense move is reversed
+            - the expense move has been reset to draft
+    The test tries to cover all the possible combinations of expense and invoicing policies, as well as the different actions
```

Comment:
```suggestion
    The test is way too complex and hard to understand. One change in this will change all the test cases and we won't understand anything to the changes done with the related PR
```